irohad/consensus/yac/storage/impl/yac_storage_cleanup_strategy_impl.cpp

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "consensus/yac/storage/yac_storage_cleanup_strategy_impl.hpp"
+
+using namespace iroha::consensus::yac;
+
+BufferedCleanupStrategy::BufferedCleanupStrategy(QueueSizeType max_limit,
+                                                 RoundType minimal_round,
+                                                 QueueType existed_state)
+    : kMaxSize_(max_limit),
+      min_round_(std::move(minimal_round)),
+      current_rounds_(std::move(existed_state)) {}
+
+boost::optional<CleanupStrategy::RoundsType> BufferedCleanupStrategy::finalize(
+    RoundType round, Answer answer) {
+  CleanupStrategy::RoundsType outcome;
+  if (current_rounds_.size() >= kMaxSize_) {
```

Comment:
```suggestion
  if (current_rounds_.size() > kMaxSize_) {
```

Reply:
Can we place an `assert` that checks `current_rounds_.size() <= kMaxSize_ + 1`? Or if we cannot, should we replace the `if` with a `while` and fill the result with all excessive rounds?